pypeit/core/telluric.py

```diff
@@ -1248,10 +1398,14 @@ def sensfunc_telluric(wave, counts, counts_ivar, counts_mask, exptime, airmass,
     telgridfile : :obj:`str`
         File containing grid of HITRAN atmosphere models; see
         :class:`~pypeit.par.pypeitpar.TelluricPar`.
+    teltype : :obj:`str`
+        Method for evaluating telluric models, either `PCA` or `grid`.
     ech_orders : `numpy.ndarray`_, shape is (norders,), optional
         If passed, provides the true order numbers for the spectra provided.
     polyorder : :obj:`int`, optional, default = 8
         Polynomial order for the sensitivity function fit.
+    ntell : :obj:`int`, optional, default = 4
+        Number of telluric model parameters (must be 4 for teltype=`grid`, or <=10 for teltype=`PCA`)
```

Comment:
I don't like this number being a free parameter that governs both. I would say just call this parameter telluric_npca or adapt it as such. Then teltype determines whether we use the fixed grid with X parameters, versus the PCA which uses telluric_npca telluric parameters.

Reply:
This took a bit more wrangling than you might have otherwise expected; there are many routines that need to know the shape of the parameter vector, and so then all those places also have to know the teltype. But it's done.